src/components/cartToggleButton/CartToggleButton.tsx

```diff
@@ -0,0 +1,49 @@
+import * as Styled from './CartToggleButton.styled';
+
+import { IMAGES } from '@/assets';
+import { AdjustQuantityButton } from '../common/adjustQuantityButton/AdjustQuantityButton';
+import { useToast } from '@/hooks/useToast';
+import { CartItemInfo } from '@/types/cartItem';
+
+interface CartItemButtonProp {
+  productId: number;
+  cartItemList?: CartItemInfo[];
+  addCartItemMutation: (productId: number) => void;
+  matchCartItem: (productId: number) => CartItemInfo | undefined;
+}
+
+const CartToggleButton = ({
+  productId,
+  cartItemList,
+  addCartItemMutation,
+  matchCartItem,
+}: CartItemButtonProp) => {
+  const { toastError } = useToast();
+
+  return (
+    <>
+      {matchCartItem(productId) ? (
+        <Styled.HandleCartItemButton $isInCart={true}>
+          <AdjustQuantityButton productId={productId} />
+        </Styled.HandleCartItemButton>
+      ) : (
+        <Styled.HandleCartItemButton
+          $isInCart={false}
+          onClick={() => {
+            if (!cartItemList) return;
+            if (cartItemList?.length >= 20) {
```

Comment:
```suggestion
            if (cartItemList.length >= 20) {
```

위에서 이미 cartItemList를 검증했으니 optional chaining은 제거해도 될거 같네요!